rotkehlchen/chain/evm/decoding/kyber/decoder.py

```diff
@@ -0,0 +1,116 @@
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any, Final
+
+from rotkehlchen.assets.asset import CryptoAsset
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.kyber.constants import CPT_KYBER
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+
+KYBER_AGGREGATOR_CONTRACT: Final = string_to_evm_address('0x6131B5fae19EA4f9D964eAc0408E4408b66337b5')  # noqa: E501
+KYBER_AGGREGATOR_TRADE: Final = b'\xd6\xd4\xf5h\x1c$l\x9fB\xc2\x03\xe2\x87\x97Z\xf1`\x1f\x8d\xf8\x03Z\x92Q\xf7\x9a\xab\\\x8f\t\xe2\xf8'  # noqa: E501
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class KyberCommonDecoder(DecoderInterface):
+
+    def _maybe_update_events(
+            self,
+            decoded_events: list['EvmEvent'],
+            sender: ChecksumEvmAddress,
+            source_asset: CryptoAsset,
+            destination_asset: CryptoAsset,
+            spent_amount: FVal,
+            return_amount: FVal,
+            counterparty: str,
+            notify_user: Callable[['EvmEvent', str], None],
+    ) -> None:
+        """
+        Use the information from a trade transaction to modify the HistoryEvents from receive/send
+        to trade if the conditions are correct.
+        """
+        in_event = out_event = None
+        for event in decoded_events:
+            try:
+                crypto_asset = event.asset.symbol_or_name()
+            except (UnknownAsset, WrongAssetType):
+                notify_user(event, counterparty)
+                continue
```

Comment:
you can let the error propagate. We do it in other decoders too. It is catch at https://github.com/rotki/rotki/blob/f352557949fe046ffd57e2e80f3459676679cd14/rotkehlchen/chain/evm/decoding/decoder.py#L364